crates/payload/util/src/traits.rs

```diff
@@ -1,23 +1,17 @@
-use std::sync::Arc;
-
-use alloy_primitives::{map::HashSet, Address};
-use reth_transaction_pool::{PoolTransaction, ValidPoolTransaction};
+use alloy_primitives::Address;
+use reth_node_builder::{BuilderContext, FullNodeTypes};
 
 /// Iterator that returns transactions for the block building process in the order they should be
 /// included in the block.
 ///
 /// Can include transactions from the pool and other sources (alternative pools,
 /// sequencer-originated transactions, etc.).
-pub trait PayloadTransactions {
+pub trait PayloadTransactions<Node: FullNodeTypes> {
     /// The transaction type this iterator yields.
     type Transaction;
 
     /// Returns the next transaction to include in the block.
-    fn next(
-        &mut self,
-        // In the future, `ctx` can include access to state for block building purposes.
-        ctx: (),
-    ) -> Option<Self::Transaction>;
+    fn next(&mut self, ctx: &BuilderContext<Node>) -> Option<Self::Transaction>;
```

Comment:
for this I think we need a new context type instead that gets populated with the block attributes, or maybe ideally with the evm as well.

but we can start with a new context type that owns block number + timestamp.

this will also we beneficial for things like conditionals